plugin/al/usdmayautils/AL/usdmaya/utils/DiffPrimVar.cpp

```diff
@@ -1674,6 +1715,78 @@ TfToken guessColourSetInterpolationTypeExtensive(
     return UsdGeomTokens->faceVarying;
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+TfToken guessColourSetInterpolationTypeExtensive(
+    const float*           rgba,
+    const size_t           numElements,
+    float                  threshold,
+    const size_t           numPoints,
+    MIntArray&             pointIndices,
+    MIntArray&             faceCounts,
+    std::vector<uint32_t>& indicesToExtract)
+{
+    // Specialized test if there is threshold provided.
+    // TODO: This function is extracted from guessColourSetInterpolationTypeExtensive() but without
+    // SSE/AVX optimization.
+    //       It is generally slower then the SSE/AVX version.
+    //       Improve it with SSE/AVX at some point when needed, for now the optimization is left for
+    //       the compiler.
+
+    if (numElements <= 1 || isWithinThreshold(rgba, numElements, threshold)) {
+        return UsdGeomTokens->constant;
+    }
+
+    float absThreshold = std::abs(threshold);
+
+    // check for per-vertex assignment
+    std::vector<uint32_t> indicesMap;
+    indicesMap.resize(numPoints, -1);
+    for (uint32_t pntInx = 0, n = pointIndices.length(); pntInx < n; ++pntInx) {
+        auto index = pointIndices[pntInx];
+        auto lastIndex = indicesMap[index];
```

Comment:
For robustness reason, I would suggest verifying that the index is within the indicesMap size, in case of data corruption.

Reply:
Done.